test/middleware/service.availability.middleware.spec.ts

```diff
@@ -109,4 +121,63 @@ describe("service availability middleware tests", () => {
 
     expect(response.text).not.toContain("Service offline - Apply to remove an overseas entity from the register");
   });
+
+  describe("Remove feature flag tests for remove specific pages", () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    test(`Redirects to the next page whenever the remove flag is true`, async () => {
+      mockIsActiveFeature
+        .mockReturnValueOnce(false) // SHOW_SERVICE_OFFLINE_PAGE
+        .mockReturnValueOnce(true) // FEATURE_FLAG_ENABLE_ROE_UPDATE
+        .mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_ROE_REMOVE
+      const resp = await request(app).get(`${REMOVE_CANNOT_USE_URL}?${PREVIOUS_PAGE_QUERY_PARAM}=${REMOVE_SOLD_ALL_LAND_FILTER_PAGE}`);
+      expect(resp.status).toEqual(302);
```

Comment:
I don't think it's enough just to check for any redirect, we should also check it is actually going to the next page, so would need a header.location check (or whatever other tests do to check it redirects to correct page)

Reply:
That works with the controller tests but this is a middleware test so I'm limited as to what I can check for, I have added what I can here.

Reply:
it's redirecting to the sign-in page, and sending user to the start of journey as they aren't logged in so just check header location is
/signin?return_to=/update-an-overseas-entity/secure-update-filter

Reply:
ignore that, the check is already there, i missed it